--- a/scripts/tools/configure
+++ b/scripts/tools/configure
@@ -147,7 +147,7 @@
 
 # additional configuration
 $configuration{"restrict-components"} = "no" unless defined $configuration{"restrict-components"};
-$configuration{"make_builder_flags"} = [ "--report-unmanaged-files" ];
+$configuration{"make_builder_flags"} = [ "--report-unmanaged-files", "--soversion=13.10" ];
 push @{$configuration{"make_builder_flags"}}, "--local-libs-only" if $configuration{"restrict-components"} eq "yes";
 $configuration{"architecture"} = $architecture;
 $configuration{"architecture-native"} = $architecture_native;
